Game/ConsoleApplication2/MovingGameObject.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "GameObject.h"
+#include "Constants.h"
+
+
+class MovingGameObject : public GameObject
+{
+public:
+	// With acceleration
+	MovingGameObject(Rect bounds, Vector2D acceleration, Vector2D maxSpeed, bool isMiniMapObject = false)
+		: GameObject(bounds),
+		m_ACCELERATION(acceleration),
+		m_MAXSPEED(maxSpeed) {}
+
+	// Without acceleration
+	MovingGameObject(Rect bounds, Vector2D maxSpeed, bool isMiniMapObject = false)
+		: GameObject(bounds),
+		m_ACCELERATION(sf::Vector2f(-1, -1)),
+		m_MAXSPEED(maxSpeed),
+		m_speed(maxSpeed) {}
+
+protected:
+	const Vector2D m_ACCELERATION;
+	const Vector2D m_MAXSPEED;
+
+	Vector2D m_direction;
+	Vector2D m_speed;
+
+
+	virtual void UpdatePosition(float dt)
+	{
+		m_bounds.pos.x += (m_direction.x * (m_speed.x * dt));
+		m_bounds.pos.y += (m_direction.y * (m_speed.y * dt));
+	}
+
+	virtual void UpdateXSpeed(float dt, int multiplier, float targetSpeed)
```

Comment:
There is no need for two functions to update something so small.

The performance impact of calling a virtual function dwarfs the doubling calculations anyway. Forcing us to call both updateX and updateY just makes the use case of updating both nearly twice as bad, while a single update is only a negligible amount faster.